docs/core/adding-spks.md

```diff
@@ -0,0 +1,683 @@
+  }
+}
+
+object MultiSigSigner extends MultiSigSigner
+```
+
+### Nested Spending Info
+
+When signing for a nested script structure, we must create a new `BitcoinSignerSingle` and a normal `BitcoinSignerFull`. For the single signer, you will likely only need to delegate with a call to `BitcoinSignerSingle.signSingle` because we only need a signature which will not likely require anything other than doing the signing on `spendingInfoToSatisfy.nestedSpendingInfo`. For the full signer, you will also need to make a delegating call with the `nestedSpendingInfo` to `BitcoinSigner.sign`, but you may also need to do whatever else is needed with the nested result to construct a correct `ScriptSignature`. For `ConditionalScriptSignature`, this all looks like:
+
+```scala
+sealed abstract class ConditionalSignerSingle
+    extends BitcoinSignerSingle[ConditionalSpendingInfoSingle] {
+
+  override def signSingle(
+      spendingInfo: UTXOSpendingInfoSingle,
+      unsignedTx: Transaction,
+      isDummySignature: Boolean,
+      spendingInfoToSatisfy: ConditionalSpendingInfoSingle)(
+      implicit ec: ExecutionContext): Future[PartialSignature] = {
+    BitcoinSignerSingle.signSingle(spendingInfo,
+                                   unsignedTx,
+                                   isDummySignature,
+                                   spendingInfoToSatisfy.nestedSpendingInfo)
+  }
+}
+
+object ConditionalSignerSingle extends ConditionalSignerSingle
+
+/** Delegates to get a ScriptSignature for the case being
+  * spent and then adds an OP_TRUE or OP_FALSE
+  */
+sealed abstract class ConditionalSigner
+    extends BitcoinSignerFull[ConditionalSpendingInfoFull] {
+
+  override def sign(
+      spendingInfo: UTXOSpendingInfoFull,
+      unsignedTx: Transaction,
+      isDummySignature: Boolean,
+      spendingInfoToSatisfy: ConditionalSpendingInfoFull)(
+      implicit ec: ExecutionContext): Future[TxSigComponent] = {
+    val (_, output, inputIndex, _) = relevantInfo(spendingInfo, unsignedTx)
+
+    val missingOpSigComponentF = BitcoinSigner.sign(
+      spendingInfo,
+      unsignedTx,
+      isDummySignature,
+      spendingInfoToSatisfy.nestedSpendingInfo)
+
+    val scriptSigF = missingOpSigComponentF.map { sigComponent =>
+      ConditionalScriptSignature(sigComponent.scriptSignature,
+                                 spendingInfoToSatisfy.condition)
+    }
+
+    updateScriptSigInSigComponent(unsignedTx,
+                                  inputIndex.toInt,
+                                  output,
+                                  scriptSigF)
+  }
+}
+object ConditionalSigner extends ConditionalSigner
+```
+
+## Step 9: Add to BitcoinSigner.sign
+
+We must now add the new signing functionality from the previous step to the general-purpose signing functions by adding a new `case` for your new `ScriptPubKey` type in the `match` within `BitcoinSigner.sign`. In the case of `P2PKWithTimeout`, this looks like:
```

Comment:
I'm not sure if this is a github bug or not but when I do `display rich diff` it shows "`ScriptPubKey` type in the `match` withing `BitcoinSigner.sign`